### ISSUE TITLE
Easy way to "freeze" BatchNorm running_mean/running_var

### ISSUE THREAD

**Issue by romain-xu-darme** [pytorch/pytorch#86590]
### 🚀 The feature, motivation and pitch

In most transfer learning applications, it is often useful to freeze some layers of the CNN (e.g. resnet50) up to the last convolutional layers in order to train only the last layers. Usually, I simply set requires_grad=False to all parameters in a simple for loop:
for param in net.parameters():
  param.requires_grad = False

This way, I can alternate freely between model.train() and model.eval(), knowing which weights will get updated. However, since batchnorm parameters (running_mean/running_var) are not updated during backprop, the requires_grad attribute set on them is pretty much ignored, contrary to the training attribute of the layer which is used to compute bn_training => after training, I end up with a model where all batchnorm params have changed.

Here's my request: could bn_training somehow also depend on the requires_grad attribute of running_mean/running_var, i.e. something like:

if self.training and self.running_mean.requires_grad and self.running_var.requires_grad:
  bn_training = True
else:
  bn_training = (self.running_mean is None) and (self.running_var is None)

Note that I've seen a lot of codes where people don't know that their feature extractor is not entirely frozen, which can lead to a lot of incorrect/non reproducible results in research papers for example.

**Comment by vadimkantorov** [comment]
Related: https://github.com/pytorch/pytorch/issues/66073 https://github.com/pytorch/pytorch/issues/68648
